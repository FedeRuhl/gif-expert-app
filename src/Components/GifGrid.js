import GifGridItem from './GifGridItem';
import PropTypes from 'prop-types';
import React from 'react';
import useFetchGifs from '../Hooks/useFetchGifs';

const GifGrid = ({category}) => {

    const { data: gifs, loading } = useFetchGifs(category);

    return (
        <>
            <h2>{ category }</h2>

            { loading && <p className="animate__animated animate__flash"> Cargando... </p> }

            <div className="card-grid">

                {
                    gifs.map((gif) => 
                    <GifGridItem 
                        key={ gif.id }
                        {...gif}
                    />)
                }
                
            </div>
        </>
    );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid
