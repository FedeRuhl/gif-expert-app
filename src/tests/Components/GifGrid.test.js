import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGrid from '../../Components/GifGrid';
import React from 'react';
import useFetchGifs from '../../Hooks/useFetchGifs';
jest.mock('../../Hooks/useFetchGifs');

describe('GifGrid component test', () => {

    const category = 'Messi';

    test('should show the component correctly', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={ category }/>);
        expect(wrapper).toMatchSnapshot();
    });
    
    test('should show items when images are loaded', () => {

        const gifs = [
            {
                id: 'ABC',
                url: 'http://test',
                title: 'ABC'
            },
            {
                id: 'ABC2',
                url: 'http://test2',
                title: 'ABC2'
            }
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={ category }/>);
        
        // expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
        
    });
    

});