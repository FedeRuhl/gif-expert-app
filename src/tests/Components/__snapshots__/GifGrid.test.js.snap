// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid component test should show the component correctly 1`] = `
<Fragment>
  <h2>
    Messi
  </h2>
  <p
    className="animate__animated animate__flash"
  >
     Cargando... 
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
